cockatrice/src/tab_supervisor.cpp

```diff
@@ -559,3 +561,19 @@ bool TabSupervisor::getAdminLocked() const
         return true;
     return tabAdmin->getLocked();
 }
+
+void TabSupervisor::processNotifyUserEvent(const Event_NotifyUser &event)
+{
+    QString notificationTitle("Notification");
+    QString notificationMessage;
+    switch ((Event_NotifyUser::NotificationType) event.type()) {
+        case Event_NotifyUser::PROMOTIONSUCCESS: QMessageBox::information(this, tr("Success"), tr("Successfully promoted user.")); break;
```

Comment:
Wait a minute... this is just an event processing function on the client side.  This data is sent from the response of the promotion in the server socket handler.  Are you confusing something here?